proposals/lip-0038.md

```diff
@@ -79,42 +80,47 @@ The Lisk protocol could contain several transaction types that perform such a re
-createAggSig(keysList, pubKeySignaturePairs):
+def createAggSig(keysList: list[bytes], pubKeySignaturePairs: list[tuple[bytes, bytes]]) -> tuple[bytes, bytes]:
     aggregationBits = byte string of length ceil(length(keyList)/8) with all bytes set to 0
     signatures = []
     for pair in pubKeySignaturePairs:
         signatures.append(pair.sig)
         index = keysList.index(pair.pubkey)
-        set bit at position index to 1 in aggregationBits
+        #set bit at position index to 1 in aggregationBits
+        aggregationBits[index // 8] |= 1 << (index % 8)
     signature = Aggregate(signatures)
     return (aggregationBits, signature)
+```
 
-verifyAggSig(keysList, aggregationBits, signature, tag, networkIdentifier, message):
-    taggedMessage = tagMessage(tag, networkIdentifier, message)
+```python
+def verifyAggSig(keysList: list[bytes], aggregationBits: bytes, signature: bytes, tag: bytes, chainID: bytes, message: bytes) -> bool:
+    taggedMessage = tagMessage(tag, chainID, message)
     keys = []
-    for every i in [0, …, ceil(length(keyList)/8)]:
-        if i-th bit of aggregationBits is set to 1:
+    for i in range(8 * len(aggregationBits)):
+        #if i-th bit of aggregationBits == 1
+        if aggregationBits[i // 8] & (1 << (i % 8)) == 1:
             keys.append(keysList[i])
-    return FastAggregateVerify(keys, taggedMessage, signature)
+    return FastAggregateVerify(keys, taggedMessage, signature) == VALID
 ```
 
 If one wants to additionally validate that the participating public keys satisfy a certain weight threshold, the function `verifyWeightedAggSig` can be used. The function takes additionally a list of weights, `weights`, where the i-th entry specifies the weight for the i-th public key in `keysList` and a weight threshold `threshold`.
 
 ```python
-verifyWeightedAggSig(keysList, aggregationBits, signature, tag, networkIdentifier, weights, threshold, message):
-    taggedMessage = tagMessage(tag, networkIdentifier, message)
+def verifyWeightedAggSig(keysList: list[bytes], aggregationBits: bytes, signature: bytes, tag: bytes, chainID: bytes, weights: list[int], threshold: int, message: bytes) -> bool:
+    taggedMessage = tagMessage(tag, chainID, message)
     keys = []
     weightSum = 0
-    for every i in [0, …, ceil(length(keyList)/8)]:
-        if i-th bit of aggregationBits is set to 1:
+    for i in range(8 * len(aggregationBits)):
+        #if i-th bit of aggregationBits == 1
+        if aggregationBits[i // 8] & (1 << (i % 8)) == 1:
             keys.append(keysList[i])
             weightSum += weights[i]
     if weightSum < threshold:
-        return INVALID
-    return FastAggregateVerify(keys, taggedMessage, signature)
+        return False
+    return FastAggregateVerify(keys, taggedMessage, signature) == VALID
 ```
 
 Note that the public keys in `pubKeySignaturePairs` need to be distinct when calling `createAggSig.` Otherwise, validation via `verifyAggSig` and `verifyWeightedAggSig` will fail.
```

Comment:
In the test vector section below there are still some occurrences of `=` instead of `==`.

Reply:
fixed